server/s3_put_object_action.cc

```diff
@@ -758,7 +758,10 @@ void S3PutObjectAction::add_object_oid_to_probable_dead_oid_list() {
   assert(!new_oid_str.empty());
 
   // store old object oid
-  if (old_object_oid.u_hi || old_object_oid.u_lo) {
+  bool delete_object =
+      (old_object_oid.u_hi || old_object_oid.u_lo) &&
```

Comment:
Why would there ever be a valid `old_object_oid` for a Put of a versioned bucket? Aren't any old version irrelevant in that case? Why even fetch the object metadata in the first place?

Reply:
Perhaps we can follow up on this and determine if it's necessary, and revert later.

Reply:
Yes, @sanalpk Please create a story ticket for this and other TODOs, if any in this PR.

Reply:
This comment will be handled in the following story:
https://jts.seagate.com/browse/EOS-26857

Resolving the conversation.